main.go

```diff
@@ -32,12 +32,12 @@ func main() {
 
 	var wg sync.WaitGroup
 	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
 
 	session, err := vici.NewSession(vici.WithSocketPath(*socketPath))
 	if err != nil {
 		log.Fatalln("error connecting to vici socket ", err)
 	}
-	defer session.Close()
```

Comment:
by adding the `defer cancel()` here it looks like we essentially flipped around the order of wg.Wait() and cancel() so that cancel() runs last. We also lost the `session.Close()` on this line. If that all sounds right, then I'm good with it.

Reply:
I was getting deadlocks in main when trying to shutdown. I moved session close into the manager avoid this defer - waitgroup interaction.

Reply:
I see it now. Perfect.